### DISCUSSION TITLE
Remove Bootstrap tooltips after up-validate

### DISCUSSION THREAD

**Discussion by mschoettle** [unpoly/unpoly#576] [Q&A] [answered]
I have a form with [radios](https://getbootstrap.com/docs/5.3/forms/checks-radios/#radios) where a Bootstrap tooltip is added on the `<label>` element of each of the radios. The corresponding `<input>` elements have `up-validate`.

The tooltips are initialized with `up.compiler`.

When the user hovers over the label and clicks on it, the old tooltip remains in the DOM because it gets added as a child of `<body>`. Once the user moves the mouse off the label this orphan tooltip remains.

What is the best way to handle this case? Would it be listening to the `up:form:validate` event and removing the tooltip manually?

**Comment by triskweline** [comment, marked as answer]
> The tooltips are initialized with up.compiler.

Your compiler could use the Bootstrap JS API to remove the tooltip on `mouseleave`. Or it could register a [destructor function](https://unpoly.com/up.compiler#destructor) that removes the tooltip when the anchor element is detached.

**Reply by mschoettle** [reply]
Thanks! It works great with the `destructor` function:

```js
up.compiler('[data-bs-toggle="tooltip"]', (element) => {
    let tooltip = new bootstrap.Tooltip(element)

    // dispose the tooltip when the anchor element is removed
    up.destructor(element, () => {
        tooltip.dispose()
    })
})
```
